db/models.js

```diff
@@ -20,18 +21,36 @@ const getArticleData = async (id) => {
   const article = await checkExists("articles", "article_id", id);
   return article.rows[0];
 };
-const getArticlesData = async () => {
+const getArticlesData = async (
+  sort_by = "created_at",
+  order = "desc",
+  topic
+) => {
+  const validOrder = ["asc", "desc"];
+  
+  if (!validOrder.includes(order)) {
+    return Promise.reject({ status: 400, msg: "400 Bad Request" });
+  }
   try {
-    const articlesComments = await db.query(`SELECT articles.*, 
-  COUNT(comment_id) 
-  AS comment_count FROM articles 
-  LEFT JOIN comments ON comments.article_id = articles.article_id 
-  GROUP BY articles.article_id
-  ORDER BY created_at DESC`);
+    let column = sort_by;
+    let queryStr = format(
+      `SELECT articles.*, 
+    COUNT(comment_id) 
+    AS comment_count FROM articles 
+    LEFT JOIN comments ON comments.article_id = articles.article_id 
+    GROUP BY articles.article_id
+    ORDER BY %I`,
+      column
+    );
+    order === "asc" ? (queryStr += ` ASC`) : (queryStr += ` DESC`);
 
-    const articlesArray = articlesComments.rows
+    const articlesComments = await db.query(queryStr);
+    const articlesArray = articlesComments.rows;
     articlesArray.forEach((article) => delete article.body);
```

Comment:
Doing this filter here isn't the most efficient way to deal with removing the body. I would recommend amending the SQL query to just _SELECT_ everything except for body. It's more characters but more efficient, imagine our production data has several million articles, that filter is going to have to run every time someone makes a request to the endpoint, which is fairly resource intense